xfrd-tcp.c

```diff
@@ -639,6 +772,80 @@ tcp_conn_ready_for_reading(struct xfrd_tcp* tcp)
 	buffer_clear(tcp->packet);
 }
 
+#ifdef HAVE_TLS_1_3
+static int
+conn_write_ssl(struct xfrd_tcp* tcp, SSL* ssl)
+{
+	ssize_t sent;
+
+	if(tcp->total_bytes < sizeof(tcp->msglen)) {
+		uint16_t sendlen = htons(tcp->msglen);
+		// send
+		int request_length = sizeof(tcp->msglen) - tcp->total_bytes;
+		sent = SSL_write(ssl, (const char*)&sendlen + tcp->total_bytes,
```

Comment:
ERR_clear_error() is called before you do SSL_read, but not here before SSL_write.

Reply:
It is in the current code, so resolved.